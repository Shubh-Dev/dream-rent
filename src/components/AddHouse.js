import { React, useState } from 'react';
import './addhouse.css';

const AddHouse = () => {
  const URL = 'https://dream-rent-api-production.up.railway.app/api/v1/houses';
  const [house, setHouse] = useState({
    address: '',
    house_type: '',
    rooms: '',
    balcony: '',
    size: '',
    image: null,
  });

  const handleChange = (e) => {
    setHouse({ ...house, [e.target.name]: e.target.value });
  };

  const redirectAfterSubmit = () => {
    setHouse({
      address: '',
      house_type: '',
      rooms: '',
      balcony: '',
      size: '',
      image: null,
    });
    window.location.replace('/');
  };

  const onImageChange = (e) => {
    setHouse({ ...house, image: e.target.files[0] });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const formData = new FormData();
    formData.append('house[address]', house.address);
    formData.append('house[house_type]', house.house_type);
    formData.append('house[rooms]', house.rooms);
    formData.append('house[balcony]', house.balcony);

    formData.append('house[size]', house.size);
    formData.append('house[image]', house.image);
    fetch(URL, {
      method: 'POST',
      body: formData,
    });
    if (house.address) redirectAfterSubmit();
  };

  return (
    <div className="add-house-form">
      <form onSubmit={handleSubmit}>
        <div className="form-group">
          <input
            type="text"
            name="address"
            placeholder="address"
            value={house.address}
            onChange={handleChange}
          />
        </div>
        <div className="form-group custom-select">
          <select name="house_type" value={house.house_type} onChange={handleChange}>
            <option value="apartment">apartment</option>
            <option value="house">house</option>
            <option value="penthouse">penthouse</option>
            <option value="studio">studio</option>
            <option value="villa">villa</option>
            <option value="igloo">igloo</option>
            <option value="treehouse">treehouse</option>
            <option value="row-house">row house</option>
            <option value="other">other</option>
          </select>
        </div>
        <div className="form-group">
          <input
            type="integer"
            name="rooms"
            placeholder="rooms"
            value={house.rooms}
            onChange={handleChange}
          />
        </div>
        <div className="form-group custom-select">
          <select name="balcony" value={house.balcony} onChange={handleChange}>
            <option value="true">true</option>
            <option value="false">false</option>
          </select>
        </div>
        <div className="form-group">
          <input
            type="integer"
            name="size"
            placeholder="size"
            value={house.size}
            onChange={handleChange}
          />
        </div>
        <div className="form-group">
          <input
            type="file"
            accept="image/*"
            name="image"
            multiple={false}
            onChange={onImageChange}
          />
        </div>
        <div className="form-group">
          <button type="submit" className="house-submit-btn" onSubmit={redirectAfterSubmit}>Submit</button>
        </div>
      </form>
    </div>
  );
};

export default AddHouse;
