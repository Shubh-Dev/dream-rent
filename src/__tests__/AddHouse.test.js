import React from "react";
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import AddHouse from "../components/AddHouse";


describe('AddHouse', () => {
    it('renders correctly', () => {
        const tree = TestRenderer.create(<AddHouse />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
    it('renders the AddHouse component', () => {
        render(<AddHouse />);
        expect(screen.getByText('Add House')).toBeInTheDocument();
    });
    }
);