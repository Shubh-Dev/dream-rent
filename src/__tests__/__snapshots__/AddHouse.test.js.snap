// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddHouse renders correctly 1`] = `
<div>
  <h2>
    Add House
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      name="address"
      onChange={[Function]}
      placeholder="address"
      type="text"
      value=""
    />
    <select
      name="house_type"
      onChange={[Function]}
      value=""
    >
      <option
        value="apartment"
      >
        apartment
      </option>
      <option
        value="house"
      >
        house
      </option>
      <option
        value="penthouse"
      >
        penthouse
      </option>
      <option
        value="studio"
      >
        studio
      </option>
      <option
        value="villa"
      >
        villa
      </option>
      <option
        value="igloo"
      >
        igloo
      </option>
      <option
        value="treehouse"
      >
        treehouse
      </option>
      <option
        value="row-house"
      >
        row house
      </option>
      <option
        value="other"
      >
        other
      </option>
    </select>
    <input
      name="rooms"
      onChange={[Function]}
      placeholder="rooms"
      type="integer"
      value=""
    />
    <select
      name="balcony"
      onChange={[Function]}
      value=""
    >
      <option
        value="true"
      >
        true
      </option>
      <option
        value="false"
      >
        false
      </option>
    </select>
    <input
      name="size"
      onChange={[Function]}
      placeholder="size"
      type="integer"
      value=""
    />
    <input
      accept="image/*"
      multiple={false}
      name="image"
      onChange={[Function]}
      type="file"
    />
    <button
      onSubmit={[Function]}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
